### PR TITLE
Add IP locker lambda.

### PR DESCRIPTION
**Pull request by MancunianSam**
Also, add a scheduled event to message once per day

### CODE REVIEW

File: ip_lock_checker_lambda.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  policies = {
+    "${local.ip_lock_checker_lambda_name}-policy" = templatefile("./templates/iam_policy/ip_lock_checker_policy.json.tpl", {
+      account_id  = var.account_number
+      lambda_name = local.ip_lock_checker_lambda_name
+    })
+  }
+  lambda_invoke_permissions = {
+    "events.amazonaws.com" = module.ip_lock_checker_cloudwatch_event.event_arn
+  }
+  memory_size = 512
```
Comment (techncl) [inline ip_lock_checker_lambda.tf:27]: Does it need this much memory?

File: templates/iam_policy/ip_lock_checker_policy.json.tpl [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "Statement": [
+    {
+      "Action": "events:PutEvents",
+      "Effect": "Allow",
+      "Resource": "arn:aws:events:eu-west-2:${account_id}:event-bus/default",
+      "Sid": "readSecretsManager"
```
Comment (techncl) [inline templates/iam_policy/ip_lock_checker_policy.json.tpl:7]: Is this name correct?

**Review by techncl — COMMENTED** [review]
Looks good; just two comments

File: ip_lock_checker_lambda.tf [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  runtime     = "python3.10"
```
Comment (techncl) [inline ip_lock_checker_lambda.tf:28]: Python 3.11 is available for Lambdas now

Comment (MancunianSam) [reply ip_lock_checker_lambda.tf:28]: I know but terraform hasn't been updated to recognise that. I've got the latest version and it still says python3.11 is invalid. We can update it once terraform catches up.

File: templates/iam_policy/ip_lock_checker_policy.json.tpl (hunk 2 above)
Comment (MancunianSam) [reply templates/iam_policy/ip_lock_checker_policy.json.tpl]: Ah no, copy and paste error. I've changed it.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**